### PR TITLE
README Improvements

### PR DESCRIPTION
**Pull request by neekolas**
## Summary

- Updates the README to include the project status (Developer Preview), and updates the status of individual components
- Adds a client development guide
- Add a section to the README about deployment

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,15 +4,17 @@ Example push notification server, written in Golang
 
 ## Project status
 
-This repo is very much a WIP, and some important features are not yet completed.
+This project is in Developer Preview. Software in this phase does is not formally supported, will change without warning, and may not be backwards compatible. We do NOT recommend using Developer Preview software in production applications.
```
Comment (elisealix22) [inline README.md:7]: Could we also add our fancy new badge for consistency? Status

**Review by elisealix22 — COMMENTED** [review]
I added @jhaaaa and @yash-luna for review just because we've been trying to sync the dev preview language across plats and want to see if they think it applies to this as well.

File: README.md (hunk 1 above)
Comment (neekolas) [reply README.md:7]: Ooh. So fancy

Comment (jhaaaa) [reply README.md:9]: Thank you @elisealix22 and @neekolas - reading this beauty TODAY! <3

File: README.md [hunk 2]
```diff
@@ -4,15 +4,19 @@ Example push notification server, written in Golang
...
+![Status](https://camo.githubusercontent.com/47c9762c88d56b96ffa436e2af994dab07f6f61f2a0388cd08be7d42b1b8fef5/68747470733a2f2f696d672e736869656c64732e696f2f62616467652f50726f6a6563745f5374617475732d446576656c6f7065725f507265766965772d79656c6c6f77)
 
-| Feature               | Status | Notes                                                        |
-| --------------------- | ------ | ------------------------------------------------------------ |
-| Installations Service | 🟢     | Some minor revisions will be needed, but looking pretty good |
-| Subscriptions Service | 🟢     |                                                              |
-| API Server            | 🟢     | Working as expected                                          |
-| XMTP Worker           | 🟡     | Needs more testing, but works                                |
-| Delivery Service      | 🟡     | Basic implementation in place. Untested                      |
+This project is in Developer Preview. Software in this phase does is not formally supported, will change without warning, and may not be backwards compatible. We do NOT recommend using Developer Preview software in production applications.
```
Comment (jhaaaa) [inline README.md:9]: What do you think of this phrasing @neekolas? @yash-luna suggested replacing the explicit warnings with softer language indicating that it is OK to start building, just not in production apps. Do you think this softer language accurately reflects where the notification server is right now - or are you more comfortable with the explicit statements? =) 

```suggestion
This project is in Developer Preview status and ready to serve as a reference for you to start building. 

However, we do NOT recommend using Developer Preview software in production apps. Software in this status may change based on feedback.
```

Comment (neekolas) [reply README.md:9]: FWIW, I got the phrasing directly from @yash-luna’s doc. Should we update the language there so we have consistency across all projects?

**Review by jhaaaa — APPROVED** [review]
Thank you so much for this documentation, Nick - I appreciate the clarity -- and the meaningful details you took the time to capture. I also appreciate the genuine voice that comes across.

I've added a few suggestions for tech writer super nit edits <3 - please feel free to use only what you find helpful.

**Review by yash-luna — APPROVED** [review]

**Pull request merged**